test/test_array_ops.cpp

```diff
@@ -902,6 +902,75 @@ TEST(ArrayOps, Rank) {
   opexecuter.RunTest();
 }  // end of RankOp
 
+// Test op: ScatterNd Op
+TEST(ArrayOps, ScatterNd1D) {
+  Tensor indices(DT_INT32, TensorShape({4, 1}));
+  Tensor updates(DT_FLOAT, TensorShape({4}));
+
+  AssignInputValues<int>(indices, {{2}, {3}, {1}, {7}});
+  AssignInputValues<float>(updates, {9.1, 10.2, -11.3, 12.4});
+
+  vector<int> static_input_indexes = {2};
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  Scope root = Scope::NewRootScope();
+  auto R = ops::ScatterNd(root, indices, updates, {8});
+  std::vector<Output> sess_run_fetchoutputs = {R};
+
+  OpExecuter opexecuter(root, "ScatterNd", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}
+
+TEST(ArrayOps, ScatterNdRepeatIndices) {
+  Tensor indices(DT_INT32, TensorShape({4, 1}));
+  Tensor updates(DT_FLOAT, TensorShape({4}));
+
+  // the index "2" appears twice
+  AssignInputValues<int>(indices, {{2}, {3}, {2}, {7}});
+  AssignInputValues<float>(updates, {9.1, 10.2, -11.3, 12.4});
+
+  vector<int> static_input_indexes = {2};
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  Scope root = Scope::NewRootScope();
+  auto R = ops::ScatterNd(root, indices, updates, {10});
+  std::vector<Output> sess_run_fetchoutputs = {R};
+
+  OpExecuter opexecuter(root, "ScatterNd", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}
+
+TEST(ArrayOps, ScatterNd3D) {
+  Tensor indices(DT_INT32, TensorShape({2, 1}));
+  Tensor updates(DT_FLOAT, TensorShape({2, 4, 4}));
+
+  AssignInputValues<int>(indices, {{0}, {2}});
+  AssignInputValuesRandom<float>(updates, -10.0, 20.0f);
+
+  vector<int> static_input_indexes = {2};
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  Scope root = Scope::NewRootScope();
+  auto R = ops::ScatterNd(root, indices, updates, {4, 4, 4});
+  std::vector<Output> sess_run_fetchoutputs = {R};
+
+  OpExecuter opexecuter(root, "ScatterNd", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}  // end of test op ScatterNd
+
+// TODO: add unit test for duplicate indices here and 3D,4D
+// these are done
+// TODO any other test? something broadcasty?
+
```

Comment:
2d to 3d? like here? https://github.com/NervanaSystems/ngraph/blob/5465677fb1abd2e8a78d4544ff2d248a7ee3ea17/test/backend/scatter.in.cpp#L255

Reply:
Tried adding those tests, but they do not seem to be right shapes for TF.